File: pyround1/83.删除排序链表中的重复元素.py
#
# @lc app=leetcode.cn id=83 lang=python
#
# [83] 删除排序链表中的重复元素
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None or head.next == None:
            return head
        p = head
        target = p.val
        while True:
            if p.next.val == target:
                p.next = p.next.next
            else:
                target = p.next.val
                p = p.next
            if p.next is None:
                break
        return head
# @lc code=end

File: cpprounttmp/83.删除排序链表中的重复元素.cpp
/*
 * @lc app=leetcode.cn id=83 lang=cpp
 *
 * [83] 删除排序链表中的重复元素
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(!head or !head->next) return head;
        auto current = head;
        while(current->next and current){
        if (current->next->val == current->val)
        {
            current->next = current->next->next;
        }
        else{
            current = current->next;
        }
        }
        return head;
    }
};
// @lc code=end

